src/runner.py

```diff
@@ -88,596 +47,416 @@ class LxdInstanceConfig(TypedDict):
-            vm_resources: Configuration of the virtual machine resources.
+    @retry(tries=10, delay=1, logger=logger)
+    def _start_instance(self, reconcile_interval) -> None:
+        """Start an instance and wait for it to boot.
 
-        Returns:
-            Difference between intended runners and actual runners.
+        Args:
+            reconcile_interval: Time in seconds of period between each reconciliation.
         """
-        runners = [r for r in self.get_info() if r.virt_type == virt_type]
-
-        # Clean up offline runners
-        offline_runners = [r for r in runners if r.is_offline]
-        if offline_runners:
-            runner_names = ", ".join(r.name for r in offline_runners)
-            logger.info(f"Cleaning up offline {virt_type} runners: {runner_names}")
-            for runner in offline_runners:
-                self._remove_runner(runner)
-
-        # Add/Remove runners to match the target quantity
-        local_online_runners = [r for r in runners if r.is_online and r.is_local]
-        delta = quantity - len(local_online_runners)
-        if delta > 0:
-            logger.info(f"Adding {delta} additional {virt_type} runner(s)")
-            for i in range(delta):
-                self.create(image="ubuntu", virt=virt_type, vm_resources=vm_resources)
-
-        if delta < 0:
-            local_idle_runners = [r for r in local_online_runners if r.is_idle]
-            # The `local_idle_runners` are filter for only local runners
-            # The if-else statement is for mypy type check.
-            local_idle_runners.sort(
-                key=lambda r: r.local.created_at if r.local is not None else False
-            )
-            offset = min(abs(delta), len(local_idle_runners))
-            if offset == 0:
-                logger.info(f"There are no idle {virt_type} runners to remove.")
-            else:
-                old_runners = local_online_runners[:offset]
-                runner_names = ", ".join(r.name for r in old_runners)
-                logger.info(f"Removing extra {offset} idle {virt_type} runner(s): {runner_names}")
-                for runner in old_runners:
-                    self._remove_runner(runner)
+        if self.instance is None:
+            return
 
-        return delta
+        logger.info("Starting LXD instance for runner: %s", self.name)
 
-    def clear(self) -> None:
-        """Clear out existing local runners."""
-        runners = [r for r in self.get_info() if r.is_local]
-        runner_names = ", ".join(r.name for r in runners)
-        logger.info(f"Removing existing local runners: {runner_names}")
-        for runner in runners:
-            self._remove_runner(runner)
+        wait_interval = 15  # seconds
+        attempts = int(reconcile_interval * 60 / wait_interval)
 
-        self._clean_unused_profiles()
+        @retry(exception=RunnerExecutionError, tries=attempts, delay=wait_interval, logger=logger)
```

Comment:
We're also nesting retries here + a sleep 60
The overall process could take a lot of time